fmriprep/utils/misc.py

```diff
@@ -96,5 +101,247 @@ def collect_bids_data(dataset, include_types=None, scan_subject='sub-',
+
+        if rec_id is not None:
+            rec_match = re.search('rec-{}_'.format(rec_id), anat_file)
+            if not rec_match:
+                continue
+
+        if file_type == 'anat':
+            if 'T1w.nii' in modality:
+                t1.append(anat_file)
+            continue
+
+    if len(t1) == 0:
+        raise NoT1sFound
+    elif len(t1) > 1:
+        raise MultipleValidT1s
+    else:
+        return t1[0]
+        
+def get_dwi(dwi_files, run_id=None, acq_id=None):
+    dwi_data = []
+    for dwi_file in dwi_files:    
+        if run_id is not None:
+            run_match = re.search('run-{}_'.format(run_id), dwi_file)
+            if not run_match:
+                continue
+
+        if acq_id is not None:
+            acq_match = re.search('acq-{}_'.format(acq_id), dwi_file)
+            if not acq_match:
+                continue
+
+        if 'sbref.nii' in modality:
+            sbref = func_file
+            bold_json = re.sub('nii(\.gz)?$', 'json', bold)
+            dwi_data.append((bold, bold_json))
+
+    if len(dwi_data) == 0:
+        raise NoDWIsFound
+    elif len(dwi_data) > 1:
+        raise MultipleValidDWIs
+    else:
+        return dwi_data[0]
+
+def get_fmap(fmap_files, run_id=None, acq_id=None):
+    return
+
+def check_intended_for(fmap_json_file, func_file):
+    try:
+        fmap_json_fp = open(fmap_json_file, 'r')
+        fmap_json = json.loads(fmap_json_fp.readlines())
+        intended_for = fmap_json["IntendedFor"]
+    except OSError:
+        return None
+    except ValueError:
+        raise InvalidFmapJsonFile
+    except KeyError:
+        return None
+    
+
+def get_ids(dataset):
+    ids = {'subjects': set(), 'sessions': set(), 'tasks': set(), 
+           'acquisitions': set(), 'reconstructions': set()}
+
+    bids_files = []
+    for path, subdirs, files in os.walk(dataset):
+        for name in files:
+            bids_files.append(os.path.join(path, name))
+
+    for bids_file in bids_files:
+        subject_match = re.search('(?<=/sub-)(.*?)(?=/)', bids_file)
+        if subject_match:
+            ids['subjects'].add(subject_match.group())
+
+        session_match = re.search('(?<=/ses-)(.*?)(?=/)', bids_file)
+        if session_match:
+            ids['sessions'].add(session_match.group())
+
+        run_match = re.search('(?<=run-)(.*?)(?=_)', bids_file)
+        if run_match:
+            ids['runs'].add(run_match.group())
+
+        acq_match = re.search('(?<=acq-)(.*?)(?=_)', bids_file)
+        if acq_match:
+            ids['acquisitions'].add(acq_match.group())
+
+        task_match = re.search('(?<=task-)(.*?)(?=_)', bids_file)
+        if task_match:
+            ids['tasks'].add(task_match.group())
+
+        rec_match = re.search('(?<=rec-)(.*?)(?=_)', bids_file)
+        if rec_match:
+            ids['reconstructions'].add(rec_match.group())
+        
+    print(ids)
+
+if __name__ == '__main__':
+    get_ids('/Users/rblair/datasets/ds172_R1.0.0')
```

Comment:
All this is a work in progress, let me remove it from the PR since its not finished or being used at the moment.